packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -190,15 +192,22 @@ export class ConstructNode {
    */
   public findAll(order: ConstructOrder = ConstructOrder.DepthFirst): IConstruct[] {
     const ret = new Array<IConstruct>();
-    const queue: IConstruct[] = [this.host];
+    visit(this.host);
+    return ret;
 
-    while (queue.length > 0) {
-      const next = order === ConstructOrder.BreadthFirst ? queue.splice(0, 1)[0] : queue.pop()!;
-      ret.push(next);
-      queue.push(...next.node.children);
-    }
+    function visit(node: IConstruct) {
+      if (order === ConstructOrder.BreadthFirst) {
```

Comment:
Btw, this is not breadth-first vs depth-first traversal. it is preorder vs postorder traversal.

Compare tree:

```
        A
       / \
      B   C
    /  \ / \
   D   E F  G
```

Breadth first (preorder): `A B C D E F G H`
Depth first preorder: `A B D E C F G`
Depth first postorder: `D E B F G C A`

Breadth first postorder: theoretically `D E F G B C A`, but people never call it that :)

Reply:
Yes, I realized that after reading your comment. I am going to revert.

Reply:
Did you need a different feature? Or did you need the reverse visiting order?

I guess the preorder/postorder might make for a more useful switch. Don't know that we really use BFS anywhere.

Reply:
I needed DFS post-order so that leaves will be synthesized before their encompassing scopes.

Reply:
It might make for a more useful default behavior, honestly. Rename the arg and leave it in?

Reply:
Or add a `childrenFirst` argument which does the reverse.